islandora_solr.api.php

```diff
@@ -177,3 +177,45 @@ function hook_islandora_solr_search_rss_item_alter($item, $doc) {
   $item['title'] = $doc['PID'];
   $item['description'] = 'this is the new rss item description';
 }
+
+/**
+ * This hook allows one to add classes to solr facet buckets (links, plus, minus).
+ *
+ * The origional intention of this hook is to allow one to implement AJAX
+ * usage on solr facets.
+ *
+ * @param array $vars
+ *   A key value pair array, containing the following:
+ *   array(
+ *     'IslandoraSolrFacets' => $object,  <- Current instance of IslandoraSolrFacets.
+ *     'bucket_attributes' => array(
+ *       'link' => array( <- Current 'link' link, its attributes and query
+ *         'attr' => &$attr,
+ *         'query' => $query_plus,
+ *       ),
+ *       'minus' => array( <- Current 'minus' link, its attributes and query
+ *         'attr' => &$attr_minus,
+ *         'query' => $query_minus,
+ *       ),
+ *       'plus' => array( <- Current 'plus' link, its attributes and query
+ *         'attr' => &$attr_plus,
+ *         'query' => $query_plus,
+ *       ),
+ *     )
+ *   );
+ */
+function hook_islandora_solr_facet_bucket_classes($vars) {
```

Comment:
Yeah... feeling like we should make up our mind, whether or not to use references... It's unusual to have references as values inside an array... Thinking it's likely to catch people off guard.